File: ias/src/main/java/com/banco/ias/business/service/CuentaService.java
package com.banco.ias.business.service;

import com.banco.ias.domain.model.Cuenta;

import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CuentaService {
    Flux<Cuenta> listar();
    Mono<Cuenta> obtenerPorNumero(String numero);
    Mono<Cuenta> findByNumero(String numero);
}

File: iasFrontCuenta/src/app/features/cuentas/services/cuenta.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../../../environments/environment';
import { Cuenta } from '../models/cuenta.model';

@Injectable({
  providedIn: 'root',
})
export class CuentaService {
  private readonly apiUrl = `${environment.apiUrl}/cuentas`;

  constructor(private readonly httpClient: HttpClient) {}

  listar(): Observable<Cuenta[]> {
    return this.httpClient.get<Cuenta[]>(this.apiUrl);
  }

  obtenerPorNumero(numero: string): Observable<Cuenta> {
    return this.httpClient.get<Cuenta>(`${this.apiUrl}/${encodeURIComponent(numero)}`);
  }
}
